packages/slate-editor/src/extensions/placeholders/PlaceholdersManager.ts

```diff
@@ -194,21 +194,35 @@ function notify<T extends Type>(type: T, uuid: Uuid, callback: (follower: Follow
     state.notifications = [...state.notifications, notification];
 
     setTimeout(() => {
-        state.notifications = state.notifications.filter((n) => n !== notification);
+        cleanupNotification(notification);
     }, NOTIFICATION_TTL);
 
     state.followers.forEach((follower) => {
         if (is(follower, type, uuid)) {
-            callback(follower as any as Follower<T>);
+            respondToNotification(follower, notification);
         }
     });
 }
 
+function cleanupNotification<T extends Type>(notification: Notification<T>) {
+    state.notifications = state.notifications.filter(
+        (n) => n !== (notification as any as Notification<Type>),
+    );
+}
+
+function respondToNotification<T extends Type>(
+    follower: Follower<T>,
+    notification: Notification<T>,
+) {
+    if (is(follower, notification.type, notification.uuid)) {
+        notification.callback(follower);
+        cleanupNotification(notification);
+    }
+}
```

Comment:
I was thinking that leaving the notification in the stack after it was processed would be problematic, so I tried to refactor the logic to clean them up right after they got processed. However, this broke the actual fix, so I reverted the changes.